From f1fa9515d6bb7d6a57a60c582b6911fd30d5e1da Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Tue, 20 Nov 2018 16:18:50 +0000
Subject: [PATCH] setup-dchroot: Request unmerged /usr

Merged /usr is known to cause multiple packages to be misbuilt. As long
as we support unmerged /usr for user systems, we should mitigate
this class of bugs by using unmerged-/usr chroots on official buildds,
resulting in binary packages that work equally well on merged- or
unmerged-/usr user systems.

See:

https://bugs.debian.org/913229
https://udd.debian.org/cgi-bin/bts-usertags.cgi?user=md@linux.it&tag=usrmerge
thread at https://lists.debian.org/debian-devel/2018/11/msg00299.html

Signed-off-by: Simon McVittie <smcv@debian.org>
---
 modules/schroot/files/setup-dchroot | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules/schroot/files/setup-dchroot b/modules/schroot/files/setup-dchroot
index c3cc504c..8a6926b3 100755
--- a/modules/schroot/files/setup-dchroot
+++ b/modules/schroot/files/setup-dchroot
@@ -352,6 +352,7 @@ debootstrap \
     --include="$include" \
     --variant=buildd \
     --arch="$arch" \
+    --no-merged-usr \
     "$suite_alias" "$rootdir" "$mirror" "$script"
 echo "$tuple" > "$rootdir/etc/debian_chroot"
 echo "force-unsafe-io" > "$rootdir/etc/dpkg/dpkg.cfg.d/force-unsafe-io"
-- 
2.19.1

